File: Algorithm/C++/35.search-insert-position.cpp
/*
 * @lc app=leetcode id=35 lang=cpp
 *
 * [35] Search Insert Position
 */

// @lc code=start
#include <vector>

class Solution {
public:
    int searchInsert(std::vector<int>& nums, int target) {
        int mid, left = 0, right = nums.size() - 1;
        while(left <= right) {
            mid = left + ((right - left) >> 1);
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return left;
    }
};
// @lc code=end

File: Algorithm/Python/35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python
#
# [35] Search Insert Position
#

# @lc code=start


class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = left + ((right-left)>>1)
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return left


        
# @lc code=end

